h/zssLogging.h

```diff
@@ -13,13 +13,570 @@
+#endif
+#define LOG_COMP_PROF_REQ_ACCESS_DEL_MSG_TEXT "profile name required for access list  DELETE\n"
+#define LOG_COMP_PROF_REQ_ACCESS_DEL          LOG_COMP_PROF_REQ_ACCESS_DEL_MSG_ID" "LOG_COMP_PROF_REQ_ACCESS_DEL_MSG_TEXT
+
+#ifndef LOG_COMP_ACCESS_REQ_ACCESS_DEL_MSG_ID
+#define LOG_COMP_ACCESS_REQ_ACCESS_DEL_MSG_ID LOG_MSG_PRFX"1418W"
+#endif
+#define LOG_COMP_ACCESS_REQ_ACCESS_DEL_MSG_TEXT "access list entry name required for access list DELETE\n"
+#define LOG_COMP_ACCESS_REQ_ACCESS_DEL        LOG_COMP_ACCESS_REQ_ACCESS_DEL_MSG_ID" "LOG_COMP_ACCESS_REQ_ACCESS_DEL_MSG_TEXT
+
+#ifndef LOG_COMP_CLASS_MGMT_QRY_DEL_MSG_ID
+#define LOG_COMP_CLASS_MGMT_QRY_DEL_MSG_ID LOG_MSG_PRFX"1419W"
+#endif
+#define LOG_COMP_CLASS_MGMT_QRY_DEL_MSG_TEXT  "class-mgmt query string is invalid, leaving...\n"
+#define LOG_COMP_CLASS_MGMT_QRY_DEL           LOG_COMP_CLASS_MGMT_QRY_DEL_MSG_ID" "LOG_COMP_CLASS_MGMT_QRY_DEL_MSG_TEXT
+
+#ifndef LOG_COMP_GROUP_REQ_PROF_POST_MSG_ID
+#define LOG_COMP_GROUP_REQ_PROF_POST_MSG_ID LOG_MSG_PRFX"1420W"
+#endif
+#define LOG_COMP_GROUP_REQ_PROF_POST_MSG_TEXT "group name required for profile POST\n"
+#define LOG_COMP_GROUP_REQ_PROF_POST          LOG_COMP_GROUP_REQ_PROF_POST_MSG_ID" "LOG_COMP_GROUP_REQ_PROF_POST_MSG_TEXT
+
+#ifndef LOG_COMP_BODY_REQ_GRP_POST_MSG_ID
+#define LOG_COMP_BODY_REQ_GRP_POST_MSG_ID LOG_MSG_PRFX"1421W"
+#endif
+#define LOG_COMP_BODY_REQ_GRP_POST_MSG_TEXT   "body not provided for group POST, leaving...\n"
+#define LOG_COMP_BODY_REQ_GRP_POST            LOG_COMP_BODY_REQ_GRP_POST_MSG_ID" "LOG_COMP_BODY_REQ_GRP_POST_MSG_TEXT
+
+#ifndef LOG_COMP_SPRR_REQ_GRP_POST_MSG_ID
+#define LOG_COMP_SPRR_REQ_GRP_POST_MSG_ID LOG_MSG_PRFX"1422W"
+#endif
+#define LOG_COMP_SPRR_REQ_GRP_POST_MSG_TEXT   "superior not provided for group POST, leaving...\n"
+#define LOG_COMP_SPRR_REQ_GRP_POST            LOG_COMP_SPRR_REQ_GRP_POST_MSG_ID" "LOG_COMP_SPRR_REQ_GRP_POST_MSG_TEXT
+
+#ifndef LOG_COMP_BSPR_PROV_GRP_POST_MSG_ID
+#define LOG_COMP_BSPR_PROV_GRP_POST_MSG_ID LOG_MSG_PRFX"1423W"
+#endif
+#define LOG_COMP_BSPR_PROV_GRP_POST_MSG_TEXT  "bad superior group provided for group POST, leaving...\n"
+#define LOG_COMP_BSPR_PROV_GRP_POST           LOG_COMP_BSPR_PROV_GRP_POST_MSG_ID" "LOG_COMP_BSPR_PROV_GRP_POST_MSG_TEXT
+
+#ifndef LOG_COMP_GROUP_REQ_USER_PP_MSG_ID
+#define LOG_COMP_GROUP_REQ_USER_PP_MSG_ID LOG_MSG_PRFX"1424W"
+#endif
+#define LOG_COMP_GROUP_REQ_USER_PP_MSG_TEXT  "group name required for user POST/PUT\n"
+#define LOG_COMP_GROUP_REQ_USER_PP           LOG_COMP_GROUP_REQ_USER_PP_MSG_ID" "LOG_COMP_GROUP_REQ_USER_PP_MSG_TEXT
+
+#ifndef LOG_COMP_ACCESS_REQ_USER_PP_MSG_ID
+#define LOG_COMP_ACCESS_REQ_USER_PP_MSG_ID LOG_MSG_PRFX"1425W"
+#endif
+#define LOG_COMP_ACCESS_REQ_USER_PP_MSG_TEXT "access type not provided for user POST/PUT, leaving...\n"
+#define LOG_COMP_ACCESS_REQ_USER_PP          LOG_COMP_ACCESS_REQ_USER_PP_MSG_ID" "LOG_COMP_ACCESS_REQ_USER_PP_MSG_TEXT
+
+#ifndef LOG_COMP_UNK_TYPE_UCCJ_MSG_ID
```

Comment:
Not a fan of needing to conditional define because of how many lines this is overall.
Also, not a fan of LOG_COMP doing double duty, where one is for a component id and another is for a log id. These are different things. Why not call these LOG_MSG_ID or something similar?

Reply:
Hey @1000TurquoisePogs , I attempted to make the structure resemble something that's already preexisting via: https://github.com/zowe/zowe-common-c/blob/fbd8d446a84186df3a996c1f0ced58dcad7717c5/h/crossmemory.h#L35

However, we do have a handful of lines of code duplication that we can maybe cut down on? I was hesitant on doing this because they're across different repos...

Reply:
What about
```
ZSS_xxxxx_MSG_ID
ZSS_xxxxx_MSG_TEXT
ZSS_xxxxx_MSG
```

Reply:
So mass remove "LOG_COMP" and make it "ZSS"? @ifakhrutdinov @1000TurquoisePogs

Reply:
@DivergentEuropeans yes, also explicitly indicate that a define is a message by having `MSG` somewhere in that `#define`'s name

Reply:
I like the ZSS_LOG idea. I'll refactor based on that